invenio_records_resources/services/events/bus.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Events bus module."""
+
+from flask import current_app
+
+
+class EventBus:
+    """Event bus."""
+
+    def __init__(self, handlers, queue):
+        """Constructor."""
+        self._handlers = handlers
+        self._queue = queue
+
+    def publish(self, event):
```

Comment:
publish should store locally events (memory), to accomodate cases of the uow failing (i.e. rollback should not keep events on the queue).

It can implement a `flush` method that will dump to the actual queue.